go/firewallsso/base.go

```diff
@@ -230,29 +239,34 @@ func (fw *FirewallSSO) logger(ctx context.Context) log15.Logger {
 // Makes sure to call FirewallSSO.Start and to validate the network and role if necessary
 func ExecuteStart(ctx context.Context, fw FirewallSSOInt, info map[string]string, timeout int) (bool, error) {
 	ctx = log.AddToLogContext(ctx, "firewall-id", fw.GetFirewallSSO(ctx).PfconfigHashNS)
-	log.LoggerWContext(ctx).Info("Processing SSO Start")
 
-	if !fw.MatchesRole(ctx, info) {
-		log.LoggerWContext(ctx).Debug(fmt.Sprintf("Not sending SSO for user device %s since it doesn't match the role", info["role"]))
-		return false, nil
-	}
+	if fw.CheckStatus(ctx, info) {
```

Comment:
To keep it a bit simpler and not nest too many conditions together, I'd suggest to apply the same recipe as with MatchesRole and MatchesNetwork which would be to just have:
```
if !fw.CheckStatus(ctx, info) {
    log.LoggerWContext(ctx).Debug("Not sending SSO for user device since it isn't registered")
    return false, nil
}
```

And that way the `else` isn't needed and adding new conditions doesn't create deeper if conditions with a new else everytime